### DISCUSSION TITLE
Deploying next js app to windows IIS server

### DISCUSSION THREAD

**Discussion by bimalmagar10** [vercel/next.js#48441] [Help]
### Summary

I have deployed my next.js app to the windows IIS server but the middleware file doesn't seem to work. The path of the pages where the middleware is matched seem to throw an internal server error. Can you please help me with this?

This is the web configuration and server.js file for deployment.
 `

          /server.js/

          const { createServer } = require('http')
          const { parse } = require('url')
          const next = require('next')
          const dev = process.env.NODE_ENV !== 'production'
          const port = process.env.PORT || 3000;
          const app = next({ dev });
          const handle = app.getRequestHandler();
          app.prepare().then(() => {
          createServer((req, res) => {
            const parsedUrl = parse(req.url, true)
            const { pathname, query } = parsedUrl
        
            if (pathname === '/a') {
              app.render(req, res, '/a', query)
            } else if (pathname === '/b') {
              app.render(req, res, '/b', query)
            } else {
              handle(req, res, parsedUrl)
            }
          }).listen(port, (err) => {
            if (err) throw err
            console.log(`> Ready on http://localhost:${port}`)
          })
        })
`

`
           
        /web.config/

           <configuration>
             <system.webServer>    
           <rewrite>
             <rules>
                 <rule name="myapp">
                     <match url="/*" />
                    <action type="Rewrite" url="server.js" />
                 </rule>
              </rules>
            </rewrite>
    
           <iisnode node_env="production" nodeProcessCommandLine="&quot;C:\Program Files\nodejs\node.exe&quot;" 
             interceptor="&quot;%programfiles%\iisnode\interceptor.js&quot;" />
    
            </system.webServer>
             <location path="" overrideMode="Deny">
                <system.webServer>
             <handlers>
             <add name="iisnode" path="server.js" verb="*" modules="iisnode" />
             </handlers>
                </system.webServer>
             </location>
        </configuration>
`

**Comment by asimplify** [comment]
Having the same issue

**Comment by asimplify** [comment]
I was able to get rid of 500 error but now the middleware is not working. Let me share my server.js file, so that you may benefit from it and maybe able to identify more solutions.

`
                                                                          /server.js/

                const { createServer } = require("http");
                const { parse } = require("url");
                const next = require("next");

                const dev = process.env.NODE_ENV !== "production";
                const hostname = "localhost";
                const port = process.env.PORT || 3000;
                const app = next({ dev, hostname, port });
                const handle = app.getRequestHandler();
  
                app.prepare().then(() => {
                 createServer(async (req, res) => {
                   try {
                      const parsedUrl = parse(req.url, true);
                      const { pathname, query } = parsedUrl;

                      if (pathname === "/a") {
                        await app.render(req, res, "/a", query);
                      } else if (pathname === "/b") {
                        await app.render(req, res, "/b", query);
                      } else {
                        await handle(req, res, parsedUrl);
                      }
                    } catch (err) {
                      console.error("Error occurred handling", req.url, err);
                      res.statusCode = 500;
                      res.end("internal server error");
                    }
                  })
                    .once("error", (err) => {
                      console.error(err);
                      process.exit(1);
                    })
                    .listen(port, () => {
                      console.log(`> Ready on http://${hostname}:${port}`);
                    });
                });
`

**Comment by CanRau** [comment]
Related https://github.com/vercel/next.js/discussions/46722, middleware is unfortunately currently edge-runtime only
